LaTeX/LaTeX.sublime-syntax

```diff
@@ -110,67 +118,111 @@ contexts:
 
   macros:
     - include: scope:text.tex#macros
-    - match: |-
-        (?x)
-        (
-          (\\)
-          (?:(?:new|renew|provide)command\*?)
-        )
-        (?:
-          (\{)(\\[A-Za-z@]+)(\})
-          | (\\[A-Za-z@])+
-        )
-        (?:(\[)(?:[^\]]*)(\]))*
-        (\{)
+    - match: '(\\)(?:new|renew|provide)command\*?'
+      scope: meta.function.latex keyword.declaration.function.latex storage.modifier.newcommand.latex
       captures:
-        1: support.function.newcommand.latex storage.modifier.newcommand.latex
-        2: punctuation.definition.backslash.latex
-        3: punctuation.definition.group.brace.begin.latex
-        4: support.function.latex entity.name.newcommand.latex
-        5: punctuation.definition.group.brace.end.latex
-        6: support.function.latex
-        7: punctuation.definition.group.bracket.begin.latex
-        8: punctuation.definition.group.bracket.end.latex
-        9: punctuation.definition.group.brace.begin.latex
+        1: punctuation.definition.backslash.latex
       push:
-        - meta_scope: meta.function.newcommand.latex
-        - match: '\}'
-          scope: punctuation.definition.group.brace.end.latex
-          pop: true
+        - newcommand-definition
+        - newcommand-optarg
+        - newcommand-argspec
+        - newcommand-commandname
+    - match: '(\\)DeclareMathOperator\*?'
+      scope: meta.function.latex support.function.declare-math-operator.latex storage.modifier.newcommand.latex
+      captures:
+        1: punctuation.definition.backslash.latex
+      push:
+        - newcommand-definition
+        - newcommand-commandname
+
+  newcommand-argspec:
```

Comment:
May the context be re-used for other function definitions? If so maybe a more common name may be useful?

Reply:
There is also the \NewDocumentCommand family of functions. These are from the xparse package, but have recently been included into the LaTeX kernel. They have a different way of specifying arguments, but the function definition and command-name parts may be the same. 
On the other hand, AFAIK, the Plain-TeX \def command actually requires that its command definition be enclosed in braces, so that is slightly different. At this point, I would keep the names as they are.